shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -41,10 +48,13 @@ - (BOOL)application:(UIApplication*)application
 
 // Returns the key window's rootViewController, if it's a FlutterViewController.
 // Otherwise, returns nil.
-+ (FlutterViewController*)rootFlutterViewController {
-  UIViewController* viewController = [UIApplication sharedApplication].keyWindow.rootViewController;
-  if ([viewController isKindOfClass:[FlutterViewController class]]) {
-    return (FlutterViewController*)viewController;
+- (FlutterViewController*)rootFlutterViewController {
+  if (_rootFlutterViewControllerGetter != nil) {
```

Comment:
Is this only used in test? It seems weird that we inject the getter this way, is it possible for the test to override this class and override this method to inject the getter there?

Reply:
I could, but it's the class I'm testing.  In order to do that I'd have to make a partial mock of the class I actually want to test which is weird, too.

I could do more traditional dependency injection and pass in an object that can get the root FlutterViewController in with the FlutterAppDelegate's initializer.  This is basically the same thing but with a setter.  I wanted to introduce the least amount of overhead possible, that's why it's a pointer check that will basically be ignored with branch prediction.  If I always passed the logic through a block or a protocol we would always incur a dynamic dispatch.